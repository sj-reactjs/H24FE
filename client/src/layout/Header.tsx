import React from 'react';
import Flex from '../components/Flex';

const Header: React.FC<any> = (props) => {
  return (
    <header>
      <Flex
        role="img"
        aria-label="Home24 Logo"
        boxShadow="md"
        backgroundColor="white"
        justifyContent="space-between"
        padding="md">
        <nav>
          <a href="/" title="Home">
            <img height="30px" src="/home24_logo.svg" alt="Home24" />
          </a>
        </nav>
        <input
          data-testid="search"
          type="text"
          placeholder={`Search Product`}
          onChange={(e) => props.changeHandler(e.target.value)} />
      </Flex>
    </header>
  );
}

export default Header;
