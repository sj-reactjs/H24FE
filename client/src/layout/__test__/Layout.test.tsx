import '@testing-library/jest-dom'
import * as React from 'react'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { screen } from '@testing-library/react'
import Footer from '../Footer'
import Header from '../Header'

describe('Header Component Test cased', () => {

  test('Header Component render', () => {
    const { asFragment, getByAltText } = render(<Header />);
    expect(getByAltText('Home24')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot();
  })

  test('Header Component > Search chaneg', () => {
    const changeHandler = jest.fn()
    render(<Header changeHandler={changeHandler} />);
    const input = screen.getByTestId('search')
    userEvent.type(input, 'test1')
    expect(input).toHaveValue('test1')
    expect(changeHandler).toHaveBeenCalled()
    userEvent.type(input, 'test2')
    expect(input).not.toHaveValue('test1')
  })
})

describe('Footer component', () => {
  test('Footer Component', () => {
    const { asFragment, getByText } = render(<Footer />);
    const footerText = `Â© Home24 ${new Date().getFullYear()}`
    expect(getByText(footerText)).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot();
  })
})