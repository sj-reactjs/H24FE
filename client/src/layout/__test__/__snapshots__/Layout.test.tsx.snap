// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Footer Component 1`] = `
<DocumentFragment>
  <footer>
    <div
      class="sc-bdvvtL sc-gsDKAQ gxSveO"
      display="flex"
    >
      <div
        class="sc-dkPtRN xjlmj"
        font-size="sm"
      >
        Â© Home24 2021
      </div>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Header Component Test cased Header Component render 1`] = `
<DocumentFragment>
  <header>
    <div
      aria-label="Home24 Logo"
      class="sc-bdvvtL sc-gsDKAQ etGSuy"
      display="flex"
      role="img"
    >
      <nav>
        <a
          href="/"
          title="Home"
        >
          <img
            alt="Home24"
            height="30px"
            src="/home24_logo.svg"
          />
        </a>
      </nav>
      <input
        data-testid="search"
        placeholder="Search Product"
        type="text"
      />
    </div>
  </header>
</DocumentFragment>
`;
